File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/details/DetailComponent.java
package com.jaynewstrom.concretesample.details;

import com.jaynewstrom.concretesample.application.ApplicationComponent;

import dagger.Component;

@ForDetailActivity
@Component(
        dependencies = {
                ApplicationComponent.class
        },
        modules = {
                DetailModule.class
        }
)
interface DetailComponent {
    void inject(DetailActivity activity);

    void inject(DetailsListItemView detailsListItemView);
}

File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/details/DetailComponent.kt
package com.jaynewstrom.concretesample.details

import com.jaynewstrom.concretesample.application.ApplicationComponent
import dagger.Component

@ForDetailActivity
@Component(dependencies = [ApplicationComponent::class], modules = [DetailModule::class])
interface DetailComponent {
    fun inject(activity: DetailActivity)
    fun inject(detailsListItemView: DetailsListItemView)
}
